### PR TITLE
Add config app and role list page

### PR DESCRIPTION
**Pull request by yuxitang-amzn**
*Issue #, if available: N/A

*Description of changes:
Add config app and role list page.

TODO: Role Table

UI Mockup: 
ss_mock

Implementation Screenshot: 
ss_impl

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/apps/configuration/panels/role-list.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          <h1>Roles</h1>
+        </EuiTitle>
+      </EuiPageHeader>
+      <EuiPageContent>
+        <EuiPageContentHeader id="role-table-container">
+          <EuiPageContentHeaderSection>
+            <EuiTitle>
+              <h3>Roles</h3>
+            </EuiTitle>
+            <EuiText size="s" color="secondary">
```
Comment (seraphjiang) [inline public/apps/configuration/panels/role-list.tsx:44]: why use secondary here, looks doesn't match with UI Mockup

File: public/apps/configuration/app-router.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  RoutesMap.permissions,
+  RoutesMap.tenets,
+  RoutesMap.auth,
+];
+
+export function AppRouter(props: AppDependencies) {
+  return (
+    <Router basename={props.params.appBasePath}>
+      <Switch>
+        <Route path={RoutesMap.roles.href}>
```
Comment (seraphjiang) [inline public/apps/configuration/app-router.tsx:78]: looks like only usage for RoutesMap is here, if so, we don't actually need to create a Map, create a RoutesList will be enough.

Comment (yuxitang-amzn) [reply public/apps/configuration/app-router.tsx:78]: If only with RoutesMap, we need to use RoutesMap[2] to access or search by id field. And in the future the `RoutesMap.users`, `RoutesMap.tenants` will also be used in the same way.

File: public/apps/configuration/panels/role-list.tsx (hunk 1 above)
Comment (yuxitang-amzn) [reply public/apps/configuration/panels/role-list.tsx]: Updated to `subdued`, screenshot updated.

File: public/apps/configuration/app-router.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+          <SplitPage
+            left={<NavPanel items={RoutesList} />}
+            right={<RoleList />}
+          />
```
Comment (mihirsoni) [inline public/apps/configuration/app-router.tsx:82]: Trying to understand the thought process here.  Is this to ensure we always have sidebar on the all the pages right ?

We could take different and simplified approach. While defining router you could also add the side bar and for the other half of the page let's have actual content. You may completely avoid the split component. 

Some reference form other plugin

https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/166cf1c6acfc8eda1429bc1d516fe1fe0c8041a8/public/pages/main/Main.tsx#L69-L117

File: public/apps/configuration/pages/split-page.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import { EuiPage, EuiPageSideBar, EuiPageBody } from '@elastic/eui';
+
+export function SplitPage(props: { left: ReactNode; right: ReactNode }) {
+  return (
+    <EuiPage>
+      <EuiPageSideBar>{props.left}</EuiPageSideBar>
+      <EuiPageBody>{props.right}</EuiPageBody>
+    </EuiPage>
+  );
+}
```
Comment (mihirsoni) [inline public/apps/configuration/pages/split-page.tsx:26]: As mentioned on consumer, we may not need this.

File: public/apps/configuration/panels/nav-panel.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      href: "#" + e.href,
+    }))
+  }]
+}
+
+export function NavPanel(props: {
+  items: RouteItem[]
+}) {
+  return (<EuiSideNav items={buildTreeItems(props.items)} />)
+}
```
Comment (mihirsoni) [inline public/apps/configuration/panels/nav-panel.tsx:36]: np: new line

File: public/apps/configuration/panels/role-list.tsx [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+          <EuiPageContentHeaderSection>
+            <EuiTitle>
+              <h3>Roles</h3>
+            </EuiTitle>
+            <EuiText size="s" color="subdued">
+              Roles are the core way of controlling access to your cluster. Roles contain any
+              combinatioin of cluster-wide permission, index-specific permissions, document- and
+              field-level security, and tenants. Then you map users to these roles so that users
+              gain those permissions. <EuiLink external={true}>Learn More</EuiLink>
+            </EuiText>
```
Comment (mihirsoni) [inline public/apps/configuration/panels/role-list.tsx:49]: I know we still haven't gone through stage of making i18n. How about we start with it and see if we can leverage Kibana's i18n ? Could you explore ?

File: public/apps/configuration/app-router.tsx (hunk 3 above)
Comment (yuxitang-amzn) [reply public/apps/configuration/app-router.tsx:82]: The side nav is only for all the listing pages, not detail pages.
ss_role_detail

File: public/apps/configuration/app-router.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
```
[the remark this answers is not in the document]
Comment (yuxitang-amzn) [reply public/apps/configuration/app-router.tsx:28]: Good catch, removed.

File: public/apps/configuration/app-router.tsx (hunk 3 above)
Comment (yuxitang-amzn) [reply public/apps/configuration/app-router.tsx]: Handled with an regex patten route.

File: public/apps/configuration/panels/role-list.tsx (hunk 6 above)
Comment (yuxitang-amzn) [reply public/apps/configuration/panels/role-list.tsx:49]: It looks not complicated. May submit a new PR to do it if have time.
https://www.elastic.co/guide/en/kibana/master/development-plugin-localization.html

**Pull request merged**